deposit/src/main/java/edu/unc/lib/deposit/transfer/TransferBinariesToStorageJob.java

```diff
@@ -164,9 +191,15 @@ private void transferDepositManifests(PID objPid, Resource resc, BinaryTransferS
                         manifestPath, getDepositUUID());
                 continue;
             }
+
+            JobStatusFactory jobStatusFactory = getJobStatusFactory();
             PID manifestPid = getDepositManifestPid(objPid, manifestFile.getName());
-            URI storageUri = transferSession.transfer(manifestPid, manifestUri);
-            resc.addLiteral(CdrDeposit.storageUri, storageUri.toString());
+
+            if (!jobStatusFactory.objectIsIngested(depositId, manifestPid.getQualifiedId())) {
```

Comment:
This seems like an operation we are going to be doing in many places. It would likely make sense and save some repetition to add protected methods to the parent class for the deposit jobs, `isObjectCompleted` and `markObjectCompleted`. They can both take a PID, the parent class should already have access to the deposit id and jobStatusFactory.

The parent class can also setup the identifier used for the progress tracking field. You should be able to do something like:
`String depositJobId = depositUUID + ":" + this.getClass().getName();`
to get the name of the job running rather than the name of parent class.